cookbook-react-native/recipes/pushNotification/index.tsx

```diff
@@ -0,0 +1,149 @@
+  channelId = defaultChannel,
+  channelName = defaultMyChannel
+}: IsetupPushNotification) {
+  PushNotification.configure({
+    //Registration token
+    onRegister: token => {
+      if (token) {
+        //save token to device
+      }
+    },
+    //this function was called when press notification
+    onNotification: notification => {
+      //When notification is open we can add a callback
+      if (handleNotification) handleNotification(notification);
+      //if notification was started by Firebase or similar in console we can add additional data to Linking or navigate ex: route: Test
+      if (notification.data.route)
+        navigationRef.current.navigate(notification.data.route);
+      notification.finish(PushNotificationIOS.FetchResult.NoData);
+    },
+    // We can add a callback if notify has action property
+    onAction: notification => {
+      console.log('ACTION:', notification.action);
+    },
+    // if we have a Error on registration this function going be called. (Warning) Error returns allways when device is a simulator IOS
+    onRegistrationError: err => {
+      // View this error here
+    },
+    //only was neccessary to IOS
+    permissions: {
+      alert: true,
+      badge: true,
+      sound: true
+    },
+    popInitialNotification: true,
+    requestPermissions: isIos
+  });
+  //PushNotification require a channel to display the notification, channelId need to be exaclty with showLocalNotification({channelId})
+  PushNotification.createChannel(
+    {
+      channelId, // (required) by default 'your-channel-id',
+      channelName, // (required) by default 'My channel'
+      channelDescription: defaultChannelDescription, // (optional) default: 'A channel to categorise your notifications'.
+      playSound: true, // (optional) default: true
+      soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
+      importance: Importance.HIGH, // (optional) default: Importance.HIGH. Int value of the Android notification importance
+      vibrate: true // (optional) default: true. Creates the default vibration pattern if true.
+    },
+    created => console.log(`createChannel returned '${created}'`) // (optional) callback returns whether the channel was created, false means it already existed.
+  );
+  return PushNotification;
+}
+export const showLocalNotification = ({
+  message,
+  title,
+  subText,
+  channelId = defaultChannel,
+  soundName,
+  bigLargeIconUrl,
+  largeIconUrl,
+  smallIcon,
+  color,
+  id = defaultChannel
+}: IlocalNotification) => {
+  if (isIos) {
+    PushNotificationIOS.addNotificationRequest({
+      id,
+      title,
+      body: message
+    });
+  } else {
+    PushNotification.localNotification({
+      message,
+      title,
+      subText,
+      channelId,
+      soundName,
+      bigLargeIconUrl,
+      smallIcon,
+      largeIconUrl,
+      color
+    });
+  }
+};
+
+const defaultDate: Date = new Date(Date.now() + 5 * 1000);
```

Comment:
wait what? this is for the delay?

Reply:
yes, I writed this delay const, so by default scheduleNotification has fireDate 5 sec of delay. Do you belive this is bad?